Customizing/global/plugins/Services/Repository/RepositoryObject/ReportOverviewVA/classes/class.ilObjReportOverviewVA.php

```diff
@@ -0,0 +1,289 @@
+			$this->s_f->reportSettings('rep_robj_ova')
+			->addSetting($this->s_f
+								->settingString('selected_study_prg', $this->plugin->txt('selected_study_prg')));
+	}
+
+	protected function getStudyId()
+	{
+		if ((string)$this->getSettingsDataFor("selected_study_prg") !== "") {
+			return (string)$this->getSettingsDataFor("selected_study_prg");
+		}
+	}
+
+	protected function getRowTemplateTitle()
+	{
+		return "tpl.gev_overview_va_row.html";
+	}
+
+	public function getRelevantParameters()
+	{
+		return $this->relevant_parameters;
+	}
+
+	protected function buildQuery($query)
+	{
+		return null;
+	}
+
+	protected function buildTable($table)
+	{
+		$table	->column("firstname", $this->plugin->txt("firstname"), true)
+				->column("lastname", $this->plugin->txt("lastname"), true)
+				->column("orgunit", $this->plugin->txt("orgunit"), true)
+				->column("entrydate", $this->plugin->txt("entrydate"), true)
+				->column("status", $this->plugin->txt("status"), true);
+
+		$osp = new ilObjStudyProgramme($this->getStudyId());
+
+		foreach ($osp->getChildren() as $child) {
+			$column_key = $child->getTitle();
+			$column_key = strtolower($column_key);
+			$column_key = str_replace(" ", "_", $column_key);
+			$table->column($column_key, $child->getTitle(), true);
+		}
+		return parent::buildTable($table);
+	}
+
+	public function filter()
+	{
+		$pf = new \CaT\Filter\PredicateFactory();
+		$tf = new \CaT\Filter\TypeFactory();
+		$f = new \CaT\Filter\FilterFactory($pf, $tf);
+		$txt = function ($id) {
+			return $this->plugin->txt($id);
+		};
+
+		return
+		$f->sequence(
+			$f->sequence(
+				$f->dateperiod(
+					$txt("period"),
+					""
+				)->map(
+					function ($start, $end) use ($f) {
+							return array (
+										  "start" => $start
+										 ,"end" => $end);
+					},
+					$tf->dict(
+						array
+							(
+								"start" => $tf->cls("DateTime")
+								,"end" => $tf->cls("DateTime")
+							)
+					)
+				),
+				$f->multiselectsearch(
+					$txt("orgunit"),
+					"",
+					$this->getOrgus()
+				),
+				$f->text(
+					$txt("lastname"),
+					""
+				),
+				$f->multiselectsearch(
+					$txt("status"),
+					"",
+					array(ilLPStatus::LP_STATUS_COMPLETED_NUM => "teilgenommen",
+							ilLPStatus::LP_STATUS_IN_PROGRESS_NUM => "in Bearbeitung",
+							ilLPStatus::LP_STATUS_FAILED_NUM => "nicht teilgenommen",
+							ilLPStatus::LP_STATUS_NOT_ATTEMPTED_NUM => "noch nicht begonnen")
+				)
+			)
+		)->map(
+			function ($start, $end, $orgunit, $lastname, $status) {
+						return array(
+									"start" => $start
+									,"end" => $end
+									,"orgunits" => $orgunit
+									,"lastname" => $lastname
+									,"status" => $status);
+			},
+			$tf->dict(
+				array(
+							 "start" => $tf->cls("DateTime")
+							 ,"end" => $tf->cls("DateTime")
+							 ,"orgunits" => $tf->lst($tf->int())
+							 ,"lastname" => $tf->string()
+							 ,"status" => $tf->lst($tf->int()))
+			)
+		);
+	}
+
+	public function buildQueryStatement()
```

Comment:
I meant to remove the function; or return null, raise an exception, whatever... this looks like pickup-leftovers.